### PR TITLE
Use SQLite CLI parameter syntax for SQLite query string

### PR DESCRIPTION
**Pull request by ajcvickers**
Part of #6482

See also https://github.com/aspnet/EntityFrameworkCore/pull/19368#issuecomment-567692210 and https://github.com/aspnet/EntityFrameworkCore/pull/19368#issuecomment-568606361

### CODE REVIEW

File: src/EFCore.Sqlite.Core/Query/Internal/SqliteQueryStringFactory.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            }
+
+            var builder = new StringBuilder();
+            foreach (DbParameter parameter in command.Parameters)
+            {
+                var value = parameter.Value;
+                builder
+                    .Append(".param set ")
+                    .Append(parameter.ParameterName)
+                    .Append(" = ")
```
Comment (bricelam) [inline src/EFCore.Sqlite.Core/Query/Internal/SqliteQueryStringFactory.cs]: Just a space here. No equals

File: src/EFCore.Sqlite.Core/Query/Internal/SqliteQueryStringFactory.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+                    .Append(
+                        value == null || value == DBNull.Value
+                            ? "NULL"
+                            : _typeMapper.FindMapping(value.GetType())?.GenerateSqlLiteral(value)
+                            ?? value.ToString())
+                    .AppendLine(";");
```
Comment (bricelam) [inline src/EFCore.Sqlite.Core/Query/Internal/SqliteQueryStringFactory.cs]: No semicolon. These aren’t actually SQL statements—just special commands handled by the sqlite3.exe tool

File: src/EFCore.Sqlite.Core/Query/Internal/SqliteQueryStringFactory.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                return command.CommandText;
+            }
+
+            var builder = new StringBuilder();
+            foreach (DbParameter parameter in command.Parameters)
+            {
+                var value = parameter.Value;
+                builder
+                    .Append(".param set ")
+                    .Append(parameter.ParameterName)
```
Comment (bricelam) [inline src/EFCore.Sqlite.Core/Query/Internal/SqliteQueryStringFactory.cs]: Needs the parameter prefix character too

File: test/EFCore.Sqlite.FunctionalTests/Query/FromSqlQuerySqliteTest.cs [hunk 4]
```diff
@@ -16,6 +17,37 @@ public FromSqlQuerySqliteTest(NorthwindQuerySqliteFixture<NoopModelCustomizer> f
...
+            Assert.Equal(expected, queryString);
+
+            return null;
+        }
+
+        public override string FromSqlRaw_queryable_with_parameters_and_closure()
+        {
+            var queryString = base.FromSqlRaw_queryable_with_parameters_and_closure();
+
+            Assert.Equal(@".param set p0 = 'London';
```
Comment (bricelam) [inline test/EFCore.Sqlite.FunctionalTests/Query/FromSqlQuerySqliteTest.cs]: Should be `.param set @p0 ‘London’`

File: src/EFCore.Sqlite.Core/Query/Internal/SqliteQueryStringFactory.cs (hunk 3 above)
Comment (ajcvickers) [reply src/EFCore.Sqlite.Core/Query/Internal/SqliteQueryStringFactory.cs]: In all the cases I tested the parameter name was already prefixed with "@". What are the cases where this is not true?

Comment (bricelam) [reply src/EFCore.Sqlite.Core/Query/Internal/SqliteQueryStringFactory.cs]: All good then—just got confused by the bad assert.

**Pull request merged**